v2/bigquery-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/templates/BigQueryToElasticsearch.java

```diff
@@ -0,0 +1,259 @@
+
+    // Create the pipeline.
+    Pipeline pipeline = Pipeline.create(options);
+
+    // Register the coders for pipeline
+    CoderRegistry coderRegistry = pipeline.getCoderRegistry();
+
+    coderRegistry.registerCoderForType(
+        FAILSAFE_ELEMENT_CODER.getEncodedTypeDescriptor(), FAILSAFE_ELEMENT_CODER);
+
+    coderRegistry.registerCoderForType(
+        FAILSAFE_TABLE_ROW_ELEMENT_CODER.getEncodedTypeDescriptor(),
+        FAILSAFE_TABLE_ROW_ELEMENT_CODER);
+
+    /*
+     * Steps: 1) Read records from BigQuery via BigQueryIO.
+     *        2) Create string from Table Row.
+     *        3) Write records to Elasticsearch.
+     *
+     *
+     * Step #1: Read from BigQuery. If a query is provided then it is used to get the TableRows.
+     */
+    PCollectionTuple convertedRows =
+        pipeline
+            .apply(
+                "ReadFromBigQuery",
+                ReadBigQuery.newBuilder()
+                    .setOptions(options.as(BigQueryToElasticsearchReadOptions.class))
+                    .build())
+            /*
+             * Step #2: Convert table rows to JSON documents.
+             */
+            .apply(
+                "TableRowsToJsonDocument",
+                TableRowToJsonDocument.newBuilder()
+                    .setOptions(options.as(JavascriptTextTransformerOptions.class))
+                    .setTransformDeadletterOutTag(TRANSFORM_DEADLETTER_OUT)
+                    .setTransformOutTag(TRANSFORM_OUT)
+                    .setUdfDeadletterOutTag(UDF_TRANSFORM_DEADLETTER_OUT)
+                    .setUdfOutTag(UDF_OUT)
+                    .build());
+
+    /*
+     * Step #3a: Write successfully converted records to Elasticsearch
+     */
+    convertedRows
+        .get(TRANSFORM_OUT)
+        .apply(
+            "GetJsonDocuments",
+            MapElements.into(TypeDescriptors.strings()).via(FailsafeElement::getPayload))
+        .apply(
+            "WriteToElasticsearch",
+            WriteToElasticsearch.newBuilder()
+                .setOptions(options.as(WriteToElasticsearchOptions.class))
+                .build());
+
+    /*
+     * Step #3b: Write records that failed conversion to BigQuery.
+     */
+    convertedRows
+        .get(TRANSFORM_DEADLETTER_OUT)
+        .apply(
+            "FailsafeTableRowsToFailsafeString", ParDo.of(new FailsafeTableRowToFailsafeStringFn()))
+        .apply(
+            "AddTimestamps",
+            WithTimestamps.of((FailsafeElement<String, String> failures) -> new Instant()))
+        .apply(
+            "WriteFailedElementsToBigQuery",
+            WriteStringMessageErrors.newBuilder()
+                .setErrorRecordsTable(
+                    ObjectUtils.firstNonNull(
+                        options.getOutputDeadletterTable(),
+                        options.getInputTableSpec() + DEFAULT_DEADLETTER_TABLE_SUFFIX))
+                .setErrorRecordsTableSchema(SchemaUtils.DEADLETTER_SCHEMA)
+                .build());
```

Comment:
Since we are only writing UDF transform failures to the deadletter table, this might not be needed.

Reply:
Has been removed.